pkg/kubelet/collector/types.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"time"
+)
+
+// These types are mostly taken directly from cadvisor to minimize the amount of changes
+// we need to do in this iteration, but these need to be cleaned up further to
+// be less cadvisor-specific and more generic to kubernetes
+
+// Simplified cadvisor.MachineInfo
+type MachineInfo struct {
+	// The number of cores in this machine.
+	NumCores int
+
+	// The amount of memory (in bytes) in this machine
+	MemoryCapacity int64
+
+	// The machine id
+	MachineID string
+
+	// The system uuid
+	SystemUUID string
+
+	// The boot id
+	BootID string
+}
+
+// Simplified cadvisor.VersionInfo
+type VersionInfo struct {
```

Comment:
Can this be a field of `MachineInfo` rather than having it's own top-level method?